File: hash.php
<?php

ini_set('memory_limit', '1G');

$i = 0;
$h = array();
while($i < 1000000) {
  $h[$i] = "aaa{$i}";
  if($i < 100) {
    echo sprintf("aaa%d", $i)."\n";
  }
  $i += 1;
}

// 1.4s, ~220MB RAM

File: hash.pl
my %h = ();
my $i = 0;

while($i < 10000000)
{
  $h{$i} = "aaa{%i}";
  
  if($i < 100) {
    print sprintf("aaa%d", $i)."\n";
  }
  $i += 1;
}

# 37s, 1.55GB RAM
